src/test/java/io/vertx/test/core/Http1xTest.java

```diff
@@ -3542,6 +3531,49 @@ public void testPerPeerPoolingWithProxy() throws Exception {
     testPerPeerPooling(i -> client.get(80, "host" + i, "/somepath"));
   }
 
+  @Test
+  public void testNoNPEWhenClientBreaksConnection() throws Exception {
+
+    final File f = setupFile("file.pdf", TestUtils.randomUnicodeString(1000000));
+
+    server.requestHandler(req -> {
+      vertx.executeBlocking(future -> {
+
+        try {
+          Thread.sleep(100);
+        } catch (InterruptedException e) {
+
+        }
+
+        future.complete();
+
+      }, false, res -> {
+        req.response().putHeader(HttpHeaders.CONTENT_TYPE, "application/pdf");
+
+        try {
+          req.response().sendFile(f.getAbsolutePath(), event -> {
+            if (event.failed()) {
+              testComplete();
+            } else {
+              fail("It should not reach this point");
+            }
+          });
+        } catch (NullPointerException e) {
```

Comment:
I think it does because we check we have a async result failure. If there was an NPE, the async result failure would not happen. What you can do perhaps instead is to use `faitFor(2)` and make the test completed by a statement after the send file and by the expected async result failure